src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java

```diff
@@ -64,6 +67,8 @@ public void fetchAndMerge(BibEntry entry, List<Field> fields) {
                 if (fetcher.isPresent()) {
                     BackgroundTask.wrap(() -> fetcher.get().performSearchById(fieldContent.get()))
                                   .onSuccess(fetchedEntry -> {
+                                      ImportCleanup cleanup = new ImportCleanup(BibDatabaseModeDetection.inferMode(new BibDatabase(List.of(entry))));
```

Comment:
Doesn't the BasePanel (`panel`) know the mode? Shouldn't that mode be used instead of guessing?